modules/common/src/main/WorkQueue.scala

```diff
@@ -49,8 +49,12 @@ final class WorkQueue(buffer: Int, timeout: FiniteDuration, name: String, parall
           case e: TimeoutException =>
             lila.mon.workQueue.timeout(name).increment()
             lila.log(s"WorkQueue:$name").warn(s"task timed out after $timeout", e)
+            promise.failure(e)
+            (task, promise): TaskWithPromise[_]
           case e: Exception =>
             lila.log(s"WorkQueue:$name").info("task failed", e)
+            promise.failure(e)
```

Comment:
I am very suspicious of this change. Why is it necessary? what even is this syntax?

Reply:
resolved in below commit.